pkg/bucketaccessrequest/bucketaccessrequest.go

```diff
@@ -0,0 +1,141 @@
+// or a special error  errBucketAccessAlreadyExists, errInvalidBucketAccessClass is returned when provisioning was impossible and
+// no further attempts to provision should be tried.
+func (b *bucketAccessRequestListener) provisionBucketAccess(ctx context.Context, bucketAccessRequest *v1alpha1.BucketAccessRequest) error {
+	bucketAccessClassName := bucketAccessRequest.Spec.BucketAccessClassName
+
+	bucketaccess := b.FindBucketAccess(ctx, bucketAccessRequest)
+	if bucketaccess != nil {
+		// bucketaccess has been already provisioned, nothing to do.
+		return util.ErrBucketAccessAlreadyExists
+	}
+
+	bucketAccessClass, err := b.bucketClient.ObjectstorageV1alpha1().BucketAccessClasses().Get(ctx, bucketAccessClassName, metav1.GetOptions{})
+	if bucketAccessClass == nil {
+		// bucket has been already provisioned, nothing to do.
+		return util.ErrInvalidBucketAccessClass
+	}
+
+	bucketRequest, err := b.bucketClient.ObjectstorageV1alpha1().BucketRequests(bucketAccessRequest.Namespace).Get(ctx, bucketAccessRequest.Spec.BucketRequestName, metav1.GetOptions{})
+	if bucketRequest == nil {
+		// bucket request does not exist, we have to reject this provision.
+		return util.ErrInvalidBucketRequest
+	}
+	if err != nil {
+		return err
+	}
+
+	if bucketRequest.Spec.BucketInstanceName == "" {
+		return util.ErrWaitForBucketProvisioning
+	}
+
+	bucketaccess = &v1alpha1.BucketAccess{}
+	bucketaccess.Name = util.GetUUID()
+
+	bucketaccess.Spec.BucketInstanceName = bucketRequest.Spec.BucketInstanceName
+	bucketaccess.Spec.BucketAccessRequest = bucketAccessRequest.Name
+	bucketaccess.Spec.ServiceAccount = bucketAccessRequest.Spec.ServiceAccountName
+	// bucketaccess.Spec.MintedSecretName - set by the driver
+	bucketaccess.Spec.PolicyActionsConfigMapData, err = util.ReadConfigData(b.kubeClient, bucketAccessClass.PolicyActionsConfigMap)
+	if err != nil {
+		return err
+	}
+	//  bucketaccess.Spec.Principal - set by the driver
+
+	bucketaccess.Spec.Provisioner = bucketAccessClass.Provisioner
+	bucketaccess.Spec.Parameters = util.CopySS(bucketAccessClass.Parameters)
+
+	bucketaccess, err = b.bucketClient.ObjectstorageV1alpha1().BucketAccesses().Create(context.Background(), bucketaccess, metav1.CreateOptions{})
+	if err == nil || apierrs.IsAlreadyExists(err) {
+		glog.V(5).Infof("BucketAccess %s saved", bucketaccess.Name)
+		return nil
+	}
+
+	glog.Infof("Finished providioning bucket access for %v", bucketAccessRequest.Name)
+	return nil
+}
+
+func (b *bucketAccessRequestListener) FindBucketAccess(ctx context.Context, br *v1alpha1.BucketAccessRequest) *v1alpha1.BucketAccess {
+	bucketAccessList, err := b.bucketClient.ObjectstorageV1alpha1().BucketAccesses().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil
+	}
+	if len(bucketAccessList.Items) > 0 {
+		for _, bucketaccess := range bucketAccessList.Items {
+			if bucketaccess.Spec.BucketAccessRequest == br.Name {
```

Comment:
according to the KEP, BA.Spec.BucketAccessRequest is an `ObjectReference. Did that change?

Reply:
Fixed